src/main/java/uk/gov/hmcts/darts/dailylist/service/impl/DailyListProcessorImpl.java

```diff
@@ -54,30 +53,42 @@ public class DailyListProcessorImpl implements DailyListProcessor {
     @Override
     @Transactional
     public void processAllDailyLists(LocalDate date) {
-        Arrays.stream(SourceType.values()).forEach(sourceType -> processDailyListForSourceType(date, sourceType));
+        processAllDailyListsForDateAndCourthouses(LocalDate.now(), courthouseRepository.findAll());
     }
 
-    private void processDailyListForSourceType(LocalDate date, SourceType sourceType) {
-        List<CourthouseEntity> allCourthouses = courthouseRepository.findAll();
-        for (CourthouseEntity courthouse : allCourthouses) {
-            List<DailyListEntity> dailyLists = dailyListRepository.findByCourthouse_IdAndStatusAndStartDateAndSourceOrderByPublishedTimestampDesc(
+    @Override
+    @Transactional
+    public void processAllDailyListForCourthouse(CourthouseEntity courthouseEntity) {
+        processAllDailyListsForDateAndCourthouses(
+            LocalDate.now(),
+            List.of(courthouseEntity)
+        );
+    }
+
+    private void processAllDailyListsForDateAndCourthouses(LocalDate date, List<CourthouseEntity> courthouseEntityList) {
+        for (CourthouseEntity courthouse : courthouseEntityList) {
+            for (SourceType source : SourceType.values()) {
+                List<DailyListEntity> dailyLists = dailyListRepository.findByCourthouse_IdAndStatusAndStartDateAndSourceOrderByPublishedTimestampDesc(
                     courthouse.getId(),
                     String.valueOf(JobStatusType.NEW),
-                    date, String.valueOf(sourceType)
-            );
-
-            // Daily lists are being ordered descending by date so first item will be the most recent version
-            if (!dailyLists.isEmpty()) {
-                try {
-                    processDailyList(dailyLists.get(0));
-                } catch (JsonProcessingException e) {
-                    dailyLists.get(0).setStatus(String.valueOf(JobStatusType.FAILED));
-                    log.error("Failed to process dailylist for courthouse: {} with dailylist id: {}", courthouse, dailyLists.get(0).getId());
+                    date, String.valueOf(source)
+                );
+                // Daily lists are being ordered descending by date so first item will be the most recent version
+                if (!dailyLists.isEmpty()) {
+                    try {
+                        processDailyList(dailyLists.get(0));
+                    } catch (JsonProcessingException e) {
+                        dailyLists.get(0).setStatus(String.valueOf(JobStatusType.FAILED));
+                        log.error("Failed to process dailylist for courthouse: {} with dailylist id: {}",
+                                  courthouse, dailyLists.get(0).getId(), e
+                        );
+                    }
+                }
+                if (dailyLists.size() > 1) {
```

Comment:
should this be inside the `if (!dailyLists.isEmpty()) {` loop, as if `dailyLists` is empty then don't need to evaluate the size?

Reply:
ACCEPT.